### ISSUE TITLE
I should be able to read encrypted scrollback in rooms, if history visibility allows.

### ISSUE THREAD

**Issue by BillCarsonFr** [vector-im/element-meta#749]
### When invited to a room I should be able to access history if visibility allows
- [ ] https://github.com/vector-im/element-meta/issues/39

### When joining a restricted room, I should be able to access history if visibility allows
- https://github.com/matrix-org/matrix-spec/issues/1
- https://github.com/vector-im/element-meta/issues/750
- https://github.com/vector-im/element-meta/issues/646

**Comment by richvdh** [comment]
Possibly related: https://github.com/vector-im/element-web/issues/24190

**Comment by MadLittleMods** [comment]
Duplicate of https://github.com/vector-im/element-meta/issues/39?

We should probably pick one or the other to use as the source of truth

---

Related Element Web issue: https://github.com/vector-im/element-meta/issues/1563

**Comment by richvdh** [comment]
> Duplicate of https://github.com/vector-im/element-meta/issues/39?

I don't think so. This is a tracking issue covering a number of pieces of work, of which vector-im/element-web#39 is one.

**Comment by joeyoropesa-dev** [comment]
Is there any news regarding this case - will we ever be able to see message history for those who are invited later..

**Comment by yangwonlohengramm** [comment]
> Is there any news regarding this case - will we ever be able to see message history for those who are invited later..

Agreed, is this just not a thing? Being able to read messages before you joined? I already tried all the room settings...

**Comment by mpeter50** [comment]
For all I know, currently if you - as a member with access to those messages - send an invite for the new person from an Element client, the new member will get the keys for the old messages when they join. I think for this to work they also need to be using an Element client, at least until the key transfer is done.
If they are already a member, they will need to first leave or be kicked.

This is not official, but I have done this successfully in the past.
Its a shame that we need to use quirks like this. Maybe someone being required to allow this could go as a feature (an optional one), but kicking and reinviting the member is very clunky.

**Comment by yangwonlohengramm** [comment]
> For all I know, currently if you - as a member with access to those messages - send an invite for the new person from an Element client, the new member will get the keys for the old messages when they join. I think for this to work they also need to be using an Element client, at least until the key transfer is done. If they are already a member, they will need to first leave or be kicked.
> 
> This is not official, but I have done this successfully in the past. Its a shame that we need to use quirks like this. Maybe someone being required to allow this could go as a feature (an optional one), but kicking and reinviting the member is very clunky.

I tried this the other day and at least in my case it didn't work, I invited someone from Element Desktop and they joined from Element on a browser and they saw all the messages  as undecryptable.

**Comment by loetsinn** [comment]
> > For all I know, currently if you - as a member with access to those messages - send an invite for the new person from an Element client, the new member will get the keys for the old messages when they join. I think for this to work they also need to be using an Element client, at least until the key transfer is done. If they are already a member, they will need to first leave or be kicked.
> > This is not official, but I have done this successfully in the past. Its a shame that we need to use quirks like this. Maybe someone being required to allow this could go as a feature (an optional one), but kicking and reinviting the member is very clunky.
> 
> I tried this the other day and at least in my case it didn't work, I invited someone from Element Desktop and they joined from Element on a browser and they saw all the messages as undecryptable.

Sadly this is still the case. Tested it this morning.
Inviting from Element Desktop. One user joining from Element Desktop one from Element Web. Both see historical messages undecryptable

**Comment by richvdh** [comment]
Correct, history sharing on invite is not supported in the element clients. This is tracked at https://github.com/element-hq/element-meta/issues/39.
